File: src/test/java/routing/RouterTest.java
package routing;

import de.esymetric.jerusalem.ownDataRepresentation.Node;
import de.esymetric.jerusalem.ownDataRepresentation.geoData.Position;
import de.esymetric.jerusalem.ownDataRepresentation.geoData.importExport.KML;
import de.esymetric.jerusalem.routing.RoutingType;

import java.io.File;
import java.util.List;
import java.util.Vector;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

public class RouterTest {

    public void makeRoute(de.esymetric.jerusalem.routing.Router router, double lat1, double lng1, double lat2,
                   double lng2, String name, String dataDirectoryPath) {
        for (RoutingType rt : RoutingType.values())
            makeRoute(router, rt.name(), lat1, lng1, lat2, lng2, name,
                    dataDirectoryPath);
    }

    void makeRoute(de.esymetric.jerusalem.routing.Router router, String routingType, double lat1, double lng1,
                   double lat2, double lng2, String name, String dataDirectoryPath) {
        System.out.println("---------------------------------------------");
        System.out
                .println("Computing Route " + name + " (" + routingType + ")");
        System.out.println("---------------------------------------------");

        List<Node> route = router
                .findRoute(routingType, lat1, lng1, lat2, lng2);
        assertNotNull(route);
        if (route == null) {
            System.out.println("ERROR: no route found for " + name + " ("
                    + routingType + ")");
            return;
        }

        System.out.println();

        KML kml = new KML();
        Vector<Position> trackPts = new Vector<Position>();
        for (Node n : route) {
            Position p = new Position();
            p.latitude = n.lat;
            p.longitude = n.lng;
            trackPts.add(p);
        }
        kml.setTrackPositions(trackPts);
        kml.Save(dataDirectoryPath + File.separatorChar + name + "-"
                + routingType + ".kml");

        assertTrue(trackPts.size() > 10);
    }
}

File: src/test/java/routing/RouterTest.kt
package routing

import de.esymetric.jerusalem.ownDataRepresentation.Node
import de.esymetric.jerusalem.ownDataRepresentation.geoData.Position
import de.esymetric.jerusalem.ownDataRepresentation.geoData.importExport.KML
import de.esymetric.jerusalem.routing.Router
import de.esymetric.jerusalem.routing.RoutingType
import org.junit.Assert
import java.io.File
import java.util.*

class RouterTest {
    fun makeRoute(
        router: Router, lat1: Double, lng1: Double, lat2: Double,
        lng2: Double, name: String, dataDirectoryPath: String
    ): Map<RoutingType, List<Node>?> {
        val map = mutableMapOf<RoutingType, List<Node>?>()
        for (rt in RoutingType.values())
            map[rt] =
            makeRoute(
            router, rt.name, lat1, lng1, lat2, lng2, name,
            dataDirectoryPath
        )
        return map
    }

    fun makeRoute(
        router: Router, routingType: String, lat1: Double, lng1: Double,
        lat2: Double, lng2: Double, name: String, dataDirectoryPath: String
    ) : List<Node>? {
        println("---------------------------------------------")
        println("Computing Route $name ($routingType)")
        println("---------------------------------------------")
        val route = router
            .findRoute(routingType, lat1, lng1, lat2, lng2)
        Assert.assertNotNull(route)
        if (route == null) {
            println(
                "ERROR: no route found for " + name + " ("
                        + routingType + ")"
            )
            return null
        }
        println()
        val kml = KML()
        val trackPts = Vector<Position>()
        for (n in route) {
            val p = Position()
            p.latitude = n.lat
            p.longitude = n.lng
            trackPts.add(p)
        }
        kml.trackPositions = trackPts
        kml.save(
            dataDirectoryPath + File.separatorChar + name + "-"
                    + routingType + ".kml"
        )
        Assert.assertTrue(trackPts.size > 10)
        return route
    }
}